cmd/ci-operator/main.go

```diff
@@ -846,10 +849,56 @@ func (o *options) writeMetadataJSON() error {
 	m.Pod = o.jobSpec.ProwJobID
 	m.WorkNamespace = o.namespace
 
-	data, err := json.MarshalIndent(m, "", "  ")
+	prowMetadataJSON, err := json.MarshalIndent(m, "", "  ")
 	if err != nil {
 		return err
 	}
+
+	// Try to find the custom prow metadata file. We assume that there's only one. If there's more than one,
+	// we'll just use the first one that we find.
+	var customProwMetadataFile string
+	filepath.Walk(o.artifactDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return filepath.SkipDir
```

Comment:
This is imho unnecessary strict and could cause a proper file being skipped. Just return nil?

https://golang.org/pkg/path/filepath/#Walk:
> If the function returns SkipDir when invoked on a non-directory file, Walk skips the remaining files in the containing directory.

Reply:
Good catch. This is probably the right way to go.